GraphQL Query Analysis feature Documentation (#1443)
wso2/docs-apim: en/docs/learn/api-security/graphql-query-analysis/query-depth-limitation.md

> Since GraphQL schemas often have circular relationships, the depth can grow till no end. 

Reviewer: ```suggestion
Since GraphQL schemas often have circular relationships, the depth can grow without bounds. 
```

> this means a client could craft a query like this:

Reviewer: ```suggestion
this means a client could craft a query as follows:
```

</p>
</div> 
</html>

### Invoke a GraphQL API

To invoke a GraphQL API, see 
[Invoke a GraphQL API]({{base_path}}/learn/consume-api/invoke-apis/invoke-apis-using-tools/invoke-an-graphql-api-using-the-integrated-graphql-console).

To perform **GraphQL Query Depth Limitation**: 

Enter the following sample query. Then click on execute button as follows.

   ```
        query{							
            allFilms{						
                id							
                species{
                    id						
                    films{
                        title					
                        planets{
                            id					
                            residents{
                                eyeColor			
                                films{
                                    director		
                                    producers
                                }
                            }
                        }
                    }
                }
            }
        }

   ```

   [![GraphQL Depth Limitation]({{base_path}}/assets/img/learn/graphql-depth-limitation-query.png)]({{base_path}}/assets/img/learn/graphql-depth-limitation-query.png)

> You have now **successfully blocked a requested GraphQL query** using the **Max Depth** value that you assigned before.

Reviewer: ```suggestion
You have now successfully blocked a GraphQL API query using the **Max Depth** value that you assigned before.
```
